climakitae/explore/typical_meteorological_year.py

```diff
@@ -0,0 +1,804 @@
+        self.vars_and_units = {
+            "Air Temperature at 2m": "degC",
+            "Dew point temperature": "degC",
+            "Relative humidity": "[0 to 100]",
+            "Instantaneous downwelling shortwave flux at bottom": "W/m2",
+            "Shortwave surface downward direct normal irradiance": "W/m2",
+            "Shortwave surface downward diffuse irradiance": "W/m2",
+            "Instantaneous downwelling longwave flux at bottom": "W/m2",
+            "Wind speed at 10m": "m s-1",
+            "Wind direction at 10m": "degrees",
+            "Surface Pressure": "Pa",
+        }
+        self.verbose = verbose
+        # These will get set later in analysis
+        self.cdf_climatology = UNSET
+        self.cdf_monthly = UNSET
+        self.weighted_fs_sum = UNSET
+        self.top_months = UNSET
+        self.all_vars = UNSET
+        self.tmy_data_to_export = UNSET
+
+    def _set_loc_from_stn_name(self, station_name: str):
+        """Get coordinates and other station metadata from station
+
+        Parameters
+        ----------
+        station_name: str
+           Name of HadISD station
+        """
+        # read in station file of CA HadISD stations
+        stn_file = pkg_resources.resource_filename(
+            "climakitae", "data/hadisd_stations.csv"
+        )
+        stn_file = pd.read_csv(stn_file, index_col=[0])
+        # grab airport
+        try:
+            self.stn_name = station_name
+            self.stn_code = stn_file.loc[stn_file["station"] == self.stn_name][
+                "station id"
+            ].item()
+            one_station = stn_file.loc[stn_file["station"] == self.stn_name]
+        except ValueError as e:
+            raise ValueError(
+                "Could  not find station in hadisd_stations.csv. Please provide valid station name."
+            ) from e
+        self.stn_lat = one_station.LAT_Y.item()
+        self.stn_lon = one_station.LON_X.item()
+        self.stn_state = one_station.state.item()
+        self._set_lat_lon()
+        return
+
+    def _set_loc_from_lat_lon(self, latitude: float, longitude: float):
+        """Set class attributes based on latitude and longitude variables.
+
+        Parameters:
+        -----------
+        latitude: float
+           Location latitude in degrees
+        longitude: float
+           Location longitude in degrees
+        """
+        self.stn_lat = latitude
+        self.stn_lon = longitude
+        self._set_lat_lon()
+        # Set station variables to string "None"
+        # to be written to file with final data
+        self.stn_name = "None"
+        self.stn_code = "None"
+        self.stn_state = "None"
+        return
+
+    def _set_lat_lon(self):
+        """Set the lat/lon ranges for selecting data around the grid point."""
+        self.latitude = (self.stn_lat - 0.05, self.stn_lat + 0.05)
+        self.longitude = (self.stn_lon - 0.06, self.stn_lon + 0.06)
+        return
```

Comment:
Out of curiosity -- is this the most efficient way of grabbing the nearest grid cell since we have a function for that?